src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly

```diff
@@ -202,20 +202,64 @@ THE SOFTWARE.
           e.onclick = function() {
             var tr = findAncestor(this,"TR");
             var inputs = tr.getElementsByTagName("INPUT");
-            for(var i=0; i&lt;inputs.length; i++){
+            for(var i=0; i < inputs.length; i++){
                 if(inputs[i].type == "checkbox") inputs[i].checked = false;
             }
+            Behaviour.applySubtree(findAncestor(this,"TABLE"),true);
             return false;
           };
           e = null; <!-- avoid memory leak -->
         });
+      Behaviour.specify("#${strategyid} td input", 'GlobalMatrixAuthorizationStrategy', 0, function(e) {
+        var impliedByString = findAncestor(e, "TD").getAttribute('data-implied-by-list');
+        var impliedByList = impliedByString.split(" ");
+        var tr = findAncestor(e,"TR");
+        e.disabled = false;
+        for (var i = 0; i < impliedByList.length; i++) {
+          var permissionId = impliedByList[i];
+          var reference = tr.querySelector("td[data-permission-id='" + permissionId + "'] input");
+          if (reference !== null) {
+            if (reference.checked) {
+              e.disabled = true;
+            }
+          }
+        }
+        e.onchange = function() {
+          var permissionId = findAncestor(this, "TD").getAttribute('data-permission-id');
+          var checked = this.checked;
+          var tr = findAncestor(this,"TR");
+          var inputs = tr.getElementsByTagName("INPUT");
+          for (var i=0; i < inputs.length; i++){
+            if(inputs[i].type == "checkbox") {
+              var td = findAncestor(inputs[i], "TD");
+              if (td !== null) {
+                var list = td.getAttribute('data-implied-by-list');
+                if (list.includes(permissionId)) {
+                  inputs[i].disabled = checked;
+                }
+              }
+            }
+          }
+          return true;
+        };
+        e = null; <!-- avoid memory leak -->
+      });
+      //]]>
         <j:if test="${empty(descriptorPath)}">
           <j:set var="descriptorPath" value="${descriptor.descriptorFullUrl}"/>
         </j:if>
+      //<![CDATA[
         <!-- validates the name -->
         Behaviour.specify("#${strategyid} TR.permission-row", 'GlobalMatrixAuthorizationStrategy', 0, function(e) {
-          FormChecker.delayedCheck("${descriptorPath}/checkName?value="+encodeURIComponent(e.getAttribute("name")),"GET",e.firstChild);
+          if (e.getAttribute('name') == '__unused__') {
+            return;
+          }
+          if (!e.hasAttribute('data-checked')) {
+            FormChecker.delayedCheck("${descriptorPath}/checkName?value="+encodeURIComponent(e.getAttribute("name")),"GET",e.firstChild);
+            e.setAttribute('data-checked', 'true');
```

Comment:
Since we add calls to `Behavior#applySubtree` on the `<table>`, not doing this attribute check would result in all rows being checked again through unrelated operations on the table. `#applySubtree` on the row doesn't work (and would still result in 1 unnecessary check each time), so do this instead.